Index: src/main/webapp/WEB-INF/jsp/users.jsp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/webapp/WEB-INF/jsp/users.jsp	(date 1562800112000)
+++ src/main/webapp/WEB-INF/jsp/users.jsp	(date 1562800135898)
@@ -1,8 +1,7 @@
 <%@ page contentType="text/html;charset=UTF-8" language="java" %>
 <%@ taglib prefix="c" uri="http://java.sun.com/jsp/jstl/core" %>
 <%@ taglib prefix="fmt" uri="http://java.sun.com/jsp/jstl/fmt" %>
-
-<fmt:setBundle basename="messages.app"/>
+<%@ taglib prefix="spring" uri="http://www.springframework.org/tags" %>
 
 <html>
 <jsp:include page="fragments/headTag.jsp"/>
@@ -10,16 +9,16 @@
 <jsp:include page="fragments/bodyHeader.jsp"/>
 
 <section>
-    <h3><fmt:message key="user.title"/></h3>
+    <h3><spring:message code="user.title"/></h3>
 
     <table border="1" cellpadding="8" cellspacing="0">
         <thead>
         <tr>
-            <th><fmt:message key="user.name"/></th>
-            <th><fmt:message key="user.email"/></th>
-            <th><fmt:message key="user.roles"/></th>
-            <th><fmt:message key="user.active"/></th>
-            <th><fmt:message key="user.registered"/></th>
+            <th><spring:message code="user.name"/></th>
+            <th><spring:message code="user.email"/></th>
+            <th><spring:message code="user.roles"/></th>
+            <th><spring:message code="user.active"/></th>
+            <th><spring:message code="user.registered"/></th>
         </tr>
         </thead>
         <c:forEach items="${users}" var="user">
Index: src/main/webapp/WEB-INF/jsp/index.jsp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/webapp/WEB-INF/jsp/index.jsp	(date 1562800112000)
+++ src/main/webapp/WEB-INF/jsp/index.jsp	(date 1562800135872)
@@ -1,8 +1,6 @@
 <%@ page contentType="text/html;charset=UTF-8" language="java" %>
 <%@ taglib prefix="c" uri="http://java.sun.com/jsp/jstl/core" %>
-<%@ taglib prefix="fmt" uri="http://java.sun.com/jsp/jstl/fmt" %>
-
-<fmt:setBundle basename="messages.app"/>
+<%@ taglib prefix="spring" uri="http://www.springframework.org/tags" %>
 
 <html>
 <jsp:include page="fragments/headTag.jsp"/>
@@ -11,11 +9,11 @@
 <br>
 <section>
     <form method="post" action="users">
-        <fmt:message key="app.login"/>: <select name="userId">
+        <spring:message code="app.login"/>: <select name="userId">
         <option value="100000" selected>User</option>
         <option value="100001">Admin</option>
     </select>
-        <button type="submit"><fmt:message key="common.select"/></button>
+        <button type="submit"><spring:message code="common.select"/></button>
     </form>
 </section>
 <jsp:include page="fragments/footer.jsp"/>
Index: src/main/webapp/WEB-INF/jsp/fragments/headTag.jsp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/webapp/WEB-INF/jsp/fragments/headTag.jsp	(date 1562800112000)
+++ src/main/webapp/WEB-INF/jsp/fragments/headTag.jsp	(date 1562800130103)
@@ -1,10 +1,9 @@
 <%@page contentType="text/html" pageEncoding="UTF-8" %>
-<%@ taglib prefix="fmt" uri="http://java.sun.com/jsp/jstl/fmt" %>
+<%@ taglib prefix="spring" uri="http://www.springframework.org/tags" %>
 <%@ taglib prefix="c" uri="http://java.sun.com/jstl/core" %>
-<fmt:setBundle basename="messages.app"/>
 
 <head>
     <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
-    <title><fmt:message key="app.title"/></title>
+    <title><spring:message code="app.title"/></title>
     <link rel="stylesheet" href="resources/css/style.css">
 </head>
\ No newline at end of file
Index: src/main/resources/messages/app_ru.properties
===================================================================
--- src/main/resources/messages/app_ru.properties	(date 1562800112000)
+++ config/messages/app_ru.properties	(date 1562799536562)
Index: src/main/resources/spring/spring-mvc.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/resources/spring/spring-mvc.xml	(date 1562800112000)
+++ src/main/resources/spring/spring-mvc.xml	(date 1562800123778)
@@ -15,4 +15,22 @@
     <bean class="org.springframework.web.servlet.view.InternalResourceViewResolver"
           p:prefix="/WEB-INF/jsp/"
           p:suffix=".jsp"/>
+
+<!--
+    <bean id="messageSource" class="org.springframework.context.support.ResourceBundleMessageSource"
+          p:cacheSeconds="60"
+          p:defaultEncoding="UTF-8">
+        <property name="basenames">
+            <list>
+                <value>messages/app</value>
+            </list>
+        </property>
+    </bean>
+-->
+    <bean id="messageSource" class="org.springframework.context.support.ReloadableResourceBundleMessageSource"
+          p:cacheSeconds="5"
+          p:defaultEncoding="UTF-8">
+        <property name="basenames" value="file:///#{systemEnvironment['TOPJAVA_ROOT']}/config/messages/app"/>
+        <property name="fallbackToSystemLocale" value="false"/>
+    </bean>
 </beans>
\ No newline at end of file
Index: src/main/webapp/WEB-INF/jsp/fragments/footer.jsp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/webapp/WEB-INF/jsp/fragments/footer.jsp	(date 1562800112000)
+++ src/main/webapp/WEB-INF/jsp/fragments/footer.jsp	(date 1562800130093)
@@ -1,5 +1,4 @@
 <%@page contentType="text/html" pageEncoding="UTF-8" %>
-<%@ taglib prefix="fmt" uri="http://java.sun.com/jsp/jstl/fmt" %>
-<fmt:setBundle basename="messages.app"/>
+<%@ taglib prefix="spring" uri="http://www.springframework.org/tags" %>
 <hr>
-<footer><fmt:message key="app.footer"/></footer>
\ No newline at end of file
+<footer><spring:message code="app.footer"/></footer>
\ No newline at end of file
Index: src/main/webapp/WEB-INF/jsp/fragments/bodyHeader.jsp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/main/webapp/WEB-INF/jsp/fragments/bodyHeader.jsp	(date 1562800112000)
+++ src/main/webapp/WEB-INF/jsp/fragments/bodyHeader.jsp	(date 1562800392573)
@@ -1,7 +1,6 @@
 <%@page contentType="text/html" pageEncoding="UTF-8" %>
-<%@ taglib prefix="fmt" uri="http://java.sun.com/jsp/jstl/fmt" %>
+<%@ taglib prefix="spring" uri="http://www.springframework.org/tags" %>
 
-<fmt:setBundle basename="messages.app"/>
 <header>
-    <a href="meals"><fmt:message key="app.title"/></a> | <a href="users"><fmt:message key="user.title"/></a> | <a href="${pageContext.request.contextPath}"><fmt:message key="app.home"/></a>
+    <a href="meals"><spring:message code="app.title"/></a> | <a href="users"><spring:message code="user.title"/></a> | <a href="${pageContext.request.contextPath}"><spring:message code="app.home"/></a>
 </header>
\ No newline at end of file
Index: src/main/resources/messages/app.properties
===================================================================
--- src/main/resources/messages/app.properties	(date 1562800112000)
+++ config/messages/app.properties	(date 1562799536556)
